### PR TITLE
Use alerting models and dtos from common-utils dependencies.

### PR DESCRIPTION
**Pull request by eirsep**
2.x PRs, whose changes are being applied to main.
https://github.com/opensearch-project/alerting/pull/577
https://github.com/opensearch-project/alerting/pull/569
https://github.com/opensearch-project/alerting/pull/568
https://github.com/opensearch-project/alerting/pull/558
https://github.com/opensearch-project/alerting/pull/556
 
**This PR also bumps Alerting plugin to 2.4**

### CODE REVIEW

**Review by sbcd90 — APPROVED** [review]

**Comment by lezzago** [comment]
Can we cherry-pick the 2.x commits and make the changes accordingly

**Comment by lezzago** [comment]
Can we reference the 2.x PRs that this is coming from?

**Comment by eirsep** [comment]
> Can we reference the 2.x PRs that this is coming from?

@lezzago have updated PR description with the same

File: build.gradle [hunk 1]
```diff
@@ -7,10 +7,10 @@ buildscript {
     apply from: 'build-tools/repositories.gradle'
 
     ext {
-        opensearch_version = System.getProperty("opensearch.version", "2.2.0-SNAPSHOT")
+        opensearch_version = System.getProperty("opensearch.version", "2.4.0-SNAPSHOT")
```
Comment (qreshi) [inline build.gradle:10]: Let's also note in the PR description that we're bumping `main` to `2.4.0` here if it's necessary for it to be in this PR. Ideally, we should separate it into another PR so it's easier to note when it happened in the commit history at a glance.

Comment (eirsep) [reply build.gradle:10]: It's necessary in this PR. Have updated the description as suggested. thanks for pointing it out.

**Review by getsaurabh02 — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (qreshi) [reply build.gradle:10]: Cool, spoke with @eirsep and we'll merge in https://github.com/opensearch-project/alerting/pull/596 and then this PR.
